src/ssl/support.cc

```diff
@@ -352,7 +352,7 @@ ssl_verify_cb(int ok, X509_STORE_CTX * ctx)
         if (error_no != SQUID_X509_V_ERR_INFINITE_VALIDATION) {
             if (check) {
                 ACLFilledChecklist *filledCheck = Filled(check);
-                assert(!filledCheck->sslErrors);
+                const auto savedErrors = filledCheck->sslErrors;
```

Comment:
AFAICT, today, nobody but this code uses the `check` object. And this code does not use the old sslErrors value (because it implements the last/current-error semantics). Thus, we do not need to save and restore old sslErrors. We can just remove the official assert(), overwrite the old value (nil or not), and then reset sslErrors to nil.

I think the current code is slightly better than blind overwrite because it relays the idea that sslErrors might become useful for other/future `check` object users. However, I will remove `savedErrors` if others think the code is better without them.

Reply:
> AFAICT, today, nobody but this code uses the `check` object. And this code does not use the old sslErrors value (because it implements the last/current-error semantics). Thus, we do not need to save and restore old sslErrors.

My logic above is flawed because the `check` object _itself_ "uses" the sslErrors value -- ACLFilledChecklist destructor cbdata-unlocks it. Thus, we need to either properly get rid of the old locked value or restore it; we cannot just forget it. The current restoring solution is better long-term (as detailed earlier), so I am closing this thread. The corresponding code in the current PR should stay as is.